### PR TITLE
Bookstore: connect to API

### PR DESCRIPTION
**Pull request by ksigei**
### In this milestone, I was able to:

1. Add redux-thunk to my project.
2. Use fetch for making API calls.
3. Fetch book list from Bookstore API on Books page load.

- Add an Action Creator, that returns a function.
- Fetch list of books.
- Dispatch another action (add an action type) that pass API response to reducer and updates the state.

4. Refactor my add book and remove book features to persist the changes in the server.   

-  Change my existing Action Creators, so they return functions.
- Make changes by using API.
- Dispatch an action (action type that existed before refactor) that updates the state accordingly.

5. Maintain the functionality after the refactor.

### CODE REVIEW

File: src/redux/configureStore.js [hunk 1]
```diff
@@ -8,8 +9,9 @@ const rootReducer = combineReducers({
 });
 
 /* eslint-disable */
-const store = createStore(rootReducer,
-window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
-);
+const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
```
Comment (thecodechaser) [inline src/redux/configureStore.js]: - [ ] There is no need to apply an extra store enhancer as you are using the default enhancer the `applymiddleware`, Please remove the `composeEnhancers`. It's also throwing an error on the console and the app is not running.

### See the image for reference 

image

**Review by thecodechaser — APPROVED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @ksigei👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/configureStore.js (hunk 1 above)
Comment (ksigei) [reply src/redux/configureStore.js]: Hi @thecodechaser, I have removed the redux dev tool config. Everything is working fine

**Review by topeogunleye — APPROVED** [review]
### Hi @ksigei, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
> 
> ## [Optional] suggestions:
> * Nothing to mention.
> 
> _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
> 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**